core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -529,6 +529,17 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
           s"(was '$javaOpts'). Use spark.executor.memory instead."
         throw new Exception(msg)
       }
+      if (javaOpts.contains("-Xms") && contains(EXECUTOR_MEMORY)) {
```

Comment:
`EXECUTOR_MEMORY` has default value, so I don't think `contains(EXECUTOR_MEMORY)` necessary condition

Reply:
On the other hand, if `spark.executor.extraJavaOptions` has this problem, `spark.driver.defaultJavaOptions(cluster deploy mode)` and `spark.yarn.am.extraJavaOptions` maybe have same problem

Reply:
> `EXECUTOR_MEMORY` has default value, so I don't think `contains(EXECUTOR_MEMORY)` necessary condition

in SparkContext Initialization process，code as follow:
    _conf = config.clone()
    _conf.validateSettings()
    ...
    ...

     _executorMemory = _conf.getOption(EXECUTOR_MEMORY.key)
      .orElse(Option(System.getenv("SPARK_EXECUTOR_MEMORY")))
      .orElse(Option(System.getenv("SPARK_MEM"))
      .map(warnSparkMem))
      .map(Utils.memoryStringToMb)
      .getOrElse(1024)
      ....


The patch will throw exception when set spark.executor.extraJavaOptions & spark.executor.memory by manual.

Reply:
> On the other hand, if `spark.executor.extraJavaOptions` has this problem, `spark.driver.defaultJavaOptions(cluster deploy mode)` and `spark.yarn.am.extraJavaOptions` maybe have same problem

Yes, subsequent patches will solve similar problems

Reply:
We should also pay attention to Kubernetes：  https://github.com/apache/spark/blob/master/resource-managers/kubernetes/docker/src/main/dockerfiles/spark/entrypoint.sh#L84-L99